### ISSUE TITLE
Feature: Ability to add Chrome extensions like password managers

### ISSUE THREAD

**Issue by MurzNN** [ferdium/ferdium-app#135]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

Rambox has ability to add extensions to the services, here is a description of this feature: https://rambox.app/feature/extensions

This is very useful to automate entering passwords using password managers like [Bitwarden](https://bitwarden.com/), also [Grammarly](https://grammarly.com/) integration will be cool too!

Shift also has the same feature: https://tryshift.com/features/extensions/

### Proposed Solution

Will be good to have same feature in Ferdium too!

### Alternatives Considered

Nope

**Comment by mattdrepo** [comment]
Not forgetting the ability to add location spoofing via a plugin. Ferdi/Ferdium allow apps like Tinder as well as other location based apps to be included - when they rely on Location.  Location is also a key part of websites serving content to people from X continent.  I would prefer location to NOT be a plugin, but when this could be the only way:  I think plugins garner more worth.

**Comment by martinszy** [comment]
Hello, this is important. Is it planned?

**Comment by TiZ-HugLife** [comment]
Hi there. Ferdium has been great so far, but this is important for me too; the organization I work for has switched to a policy that invalidates active logins every day, which means that without an autofilling password manager, I have to freshly log in to all of my apps that use my org's credentials *every day*. This causes me to risk missing important notifications from my job's Teams chats, as one example.

**Comment by cramberries** [comment]
Would love to have this feature as well. 

If I just may add another use-case to the benefits of supporting extensions: being able to use scripts on certain websites (via extensions like TamperMonkey/GreaseMonkey) would make a lot of things easier for some sites as well.

**Comment by bassmanitram** [comment]
I just wanna disable auto modification of sound device volumes - there's an extension for that, so this issue would certainly help.

**Comment by Fijxu** [comment]
+1. I would love to use Ublock Origin on Ferdium so Discord and all the other services don't send sketchy telemetry data to they servers

**Comment by rndthoughts** [comment]
> Hi there. Ferdium has been great so far, but this is important for me too; the organization I work for has switched to a policy that invalidates active logins every day, which means that without an autofilling password manager, I have to freshly log in to all of my apps that use my org's credentials _every day_. This causes me to risk missing important notifications from my job's Teams chats, as one example.

This, times 100.

**Comment by soumyaDghosh** [comment]
> +1. I would love to use Ublock Origin on Ferdium so Discord and all the other services don't send sketchy telemetry data to they servers

Same here

**Comment by Havock94** [comment]
+1 and following. I also use Ferdium for custom websites (news, forums etc) and some of them have really invasive ads, can't wait for extensions support!

**Comment by RJVB** [comment]
+++ for adblocking, now that Instagram is also serving ads to its "desktop" version!

**Comment by siamak-haschemi** [comment]
+1 My use-case is also to use a password-manager extension. I use KeePassXC on my mac, and a browser-plugin which connects to the locally running application to get the username/passwords for entering in the browser. I would love to be able to install the chrome-browser-plugin in Ferdium.

**Comment by wallieee** [comment]
**+1** Looking forward for this to be implemented, extensions are really useful on the web apps such as ad blockers, etc...

**Comment by appel** [comment]
Not to further bloat this issue, as I'm sure the devs are aware of the upsides of extension support. Just want to point out that besides the obvious use case for extensions like adblockers and password managers I'd also like to be able to use extensions that enhance a specific service. Some are well-known, others are quite "niche" and specific.

For example, I use ["Gmail Simple Notes"](https://chrome.google.com/webstore/detail/simple-gmail-notes/jfjkcbkgjohminidbpendlodpfacgmlm) for adding notes to emails. I also use ["Gmail Quick Links"](https://chrome.google.com/webstore/detail/gmail-quick-links/ecbkcjeoffcjnppapdlncohmehhnfibd) to store custom searches. And I use ["Stylus"](https://chrome.google.com/webstore/detail/stylus/clngdbkpkpeebahjckkjfobafhncgmne) across my services to clean up the interface.

Despite Ferdium already being super useful, not having access to these extensions are currently keeping me from fully switching over from my cluttered SSB setup.

**Comment by RJVB** [comment]
On a more general note: certain really useful extensions (like FB Purity) also exist as userscripts to be run in Grease/TamperMonkey. It should be possible to run those as a `user.js` script in a specific service but there are a few problems with that:

- it's not clear how to inject the extension code into the template user.js script
- the extensions do not seem to have access to presenting a configuration dialog and/or storage for their settings. FB Purity for instance always thinks it's being run for the 1st time
- you can only add a single extension (presumably!) which gets stored in the `recipes` directory. There's only 1 such directory for all custom websites you might add. True extensions would presumably be stored under the `partitions` directory, which has individual entries for each service.

One thing the devs may want to know is if we expect extensions (and their settings!) to be synched... Personally I'd consider it progress already if they're not - I don't expect to be installing as many extensions as I have in my browser.

**Comment by lemikeone** [comment]
I would really like to have the ability to install Chrome extensions too, especially to use [LanguageTool](https://languagetool.org/fr).

**Comment by ThinkSalat** [comment]
I've also made this comment in the discord. Because it runs on chromium it should be easy to allow extensions. Many of Ferdi's competitors in the all in one services window space allow extensions already. not sure why this hasn't been a day one feature. Have any developers chimed in as to why this feature is still blocked?

**Comment by lemikeone** [comment]
@ThinkSalat, which competitors of Ferdium are you referring to? 
I have tried several in recent days, including Franz, texts.com, Beeper, Rambox, and Singlebox, but none of them support Chrome extensions. 
The only options that do support extensions are more like alternative browsers, providing more features than I require. 
Currently, Singlebox works best for me. I would have preferred to stay with Ferdium, but I often encountered issues with the purple splash screen freeze and slow loading times. 
If Ferdium were to add support for Chrome extensions, I would definitely reconsider my choice.

**Comment by Havock94** [comment]
@lemikeone I haven't got much experience with the alternatives but Rambox claims to support extensions in its website (although in a pro plan)

**Comment by lemikeone** [comment]
@Havock94 For what I have seen when I tried it, they only support their own selection of extensions (LanguageTool is not part of it)  : 1Password, BitWarden, Boomerang for Gmail, Dark Mode, Dark Reader, DashLane, DragApp, Google Docs Offline, Google Translate, Grammarly, LastPass, NordPass, SafelnCloud and uBlock.

**Comment by MurzNN** [comment]
[Shift](https://tryshift.com/extension/) also provides wide list of extensions:  https://tryshift.com/extension/

**Comment by lemikeone** [comment]
Thanks @MurzNN, I had a look at Shift, but I have to say that $149 per year to manage my messaging apps is a bit too much for me 😄

**Comment by ThinkSalat** [comment]
Yeah, was in love with Shift till I say $149 price tag PER YEAR. these people are psychos. We can make Ferdium better and completely free. There are a ton of cool apps like shift that cost less and have the same features. 

@lemikeone It's true, oftentimes these electron apps have somewhat limited extension pools or are not well tested.

**Comment by PandaWhisperer** [comment]
Quick tip for those of you wanting support for blocking ads in Ferdium: you CAN block (or at least hide) advertisements using custom rules in your `user.css`. 

For instance, adding this rule in `user.css` for Twitter will block promoted tweets in your timeline:

    [data-testid="cellInnerDiv"] [data-testid="placementTracking"]:has([data-testid*="impression-pixel"]) { display: none; }

It's a bit of a PITA to figure out these rules and requires decent knowledge of CSS and use of the web inspector, but it does work. For more complex cases that aren't possible to target with CSS alone, there's also a `user.js` that allows injecting custom JavaScript into the page.

**Comment by Havock94** [comment]
In my case, as a web developer, I have no problem figuring out the CSS rules for the websites I'm using, but unfortunately some websites use dynamic classes for their ads containers and that's impossible to manage with CSS only.
Also remember that hiding the ads containers doesn't prevent the actual HTTP requests, that can slow down the page or even download malwares or adwares.
That's where an adBlock comes handy.

CSS rules are of course a temporary workaround to hide the disturbing containers while we wait for extensions support, but it's not a "definitive" solution.

**Comment by RJVB** [comment]
On Thursday February 23 2023 01:28:55 Luca wrote:

>That's where an adBlock comes handy.

Isn't a browser extension essentially a userscript with access to features that make it easier to use but do not concern its core function? Something like FB Purity exists as a userscript and that version does do its job in Ferdium. The only thing that doesn't work is setting the configuration options.

BTW: https://www.electronjs.org/docs/latest/api/extensions

**Comment by Havock94** [comment]
@RJVB You're correct, but extensions can also access the browser APIs in ways you wouldn't be able to with simple scripts.
I always use the example of an Ad Block, where the source code can contain thousands of code lines. You can write your own AdBlock, but why reinventing the wheel when it's already been done?
Also by implementing extensions you'd also have automatic updates without having to worry to manually update the code on every release, let alone the fact that not everyone has the knowledge to write or even find and install scripts.

Again, with pure CSS and pure JS you can definitely do something, native extensions support is the next step

**Comment by martinszy** [comment]
So... if user scripts can be used to add functionality... maybe they could be used to write the extension manager?

**Comment by miztizm** [comment]
this would be a killer function +1111111111111111111111111111111111111111111111

**Comment by mcopik** [comment]
:+1: Integration of extensions such as Grammarly would be perfect. Right now, I can't make a complete switch with my mail client from browser to Ferdium since I need spellchecking and grammar verification for writing emails.

**Comment by bboysnick5** [comment]
Please make it happen. begging.

**Comment by benzntech** [comment]
Here are some references done by [getstation](https://getstation.com/) for the chrome-extension 
https://github.com/getstation/electron-chrome-extension

**Comment by RJVB** [comment]
As far as ad-blocking is concerned it seems uBlock Origin could be integrated via  the npm/nodejs module they provide:

https://github.com/gorhill/uBlock/tree/master/platform/nodejs

**Comment by wokawoka** [comment]
ublock origin integration would be a gamechanger!

**Comment by katastrophe92** [comment]
I use NextDNS or the NordVPN adblocking feature when VPN is ON. A lot of VPN providers have nowadays adblock feature. But NextDNS is the best. Even the free plan is enough for most peoples. I even have it as a "custom recipe" in Ferdium to look at the logs if something is blocked that I need, if is a false positive or to quick change something. It is really cheap the year plan and is working on every device
If Ferdium would have extensions and maybe go away fron electron it will be the best. But I still love it 😍

**Comment by Cris70** [comment]
> @ThinkSalat, which competitors of Ferdium are you referring to? 

Wavebox supports all Chrome extensions.

**Comment by miztizm** [comment]
> > @ThinkSalat, which competitors of Ferdium are you referring to?
> 
> Wavebox supports all Chrome extensions.

its not a freemium software

**Comment by ThinkSalat** [comment]
Ferdium still doesn’t have extensions?

**Comment by benzntech** [comment]
[getstation](https://getstation.com/)  supports all Chrome extensions. 

>This project is not actively maintained. Forks or PRs encouraged (PRs will be reviewed)

**Comment by Cris70** [comment]
> > > @ThinkSalat, which competitors of Ferdium are you referring to?
> > 
> > 
> > Wavebox supports all Chrome extensions.
> 
> its not a freemium software

Mmmm why do you say that?
It is free for basic usage, with paid plans for advanced usage.
See https://wavebox.io/pricing.

**Comment by Cris70** [comment]
> [getstation](https://getstation.com/) supports all Chrome extensions.

Does not seem to be the case: see https://github.com/getstation/desktop-app/wiki/FAQ-%7C-%F0%9F%93%B1-Applications-&-extensions#can-i-use-extensions-inside-station

**Comment by miztizm** [comment]
> > > > @ThinkSalat, which competitors of Ferdium are you referring to?
> > > 
> > > 
> > > Wavebox supports all Chrome extensions.
> > 
> > 
> > its not a freemium software
> 
> Mmmm why do you say that? It is free for basic usage, with paid plans for advanced usage. See https://wavebox.io/pricing.

when i add a extensions it says i need to pay.

**Comment by MurzNN** [comment]
I've found this library for Electron from Rambox developers: https://github.com/ramboxapp/electron-chrome-extensions
Can we use it in Ferdium in the same way?

**Comment by martinszy** [comment]
> I've found this library for Electron from Rambox developers: https://github.com/ramboxapp/electron-chrome-extensions Can we use it in Ferdium in the same way?

That looks  promising. Maybe someone with the dev environment setup should try inserting the library at this location: https://github.com/ferdium/ferdium-app/blob/develop/src/index.ts#LL77C5-L77C7

The code would be this one: https://github.com/ramboxapp/electron-chrome-extensions/blob/master/README.md#advanced

Hopefully it's that simple. But I doubt it because ferdium has many custom behaviors.

**Comment by RJVB** [comment]
>That looks  promising. Maybe someone with the dev environment setup should try inserting the library at this location: https://github.com/ferdium/ferdium-app/blob/develop/src/index.ts#LL77C5-L77C7

Or just anyone who builds Ferdium from source...

**Comment by josefpsurny** [comment]
Can we have your attention @vraravam  please and review possible implementation of this feature which can be done easily as show other similar apps to Ferdium? Thanks a lot!

**Comment by vraravam** [comment]
@josefpsurny - i have already announced that I will not be contributing actively to develop features in Ferdium. Please do not tag me again in any issue

**Comment by benzntech** [comment]
> [getstation](https://getstation.com/) supports all Chrome extensions.
> 
> > This project is not actively maintained. Forks or PRs encouraged (PRs will be reviewed)

getstation is active now @MurzNN @vraravam

**Comment by RJVB** [comment]
>getstation is active now @MurzNN @vraravam 

But see https://github.com/getstation/desktop-app/issues/131 !

Wherein I also learnt about Biscuit (eatbiscuit.com) but apparently that isn't well maintained and doesn't support extensions either.

**Comment by quentingosset** [comment]
> > I've found this library for Electron from Rambox developers: https://github.com/ramboxapp/electron-chrome-extensions Can we use it in Ferdium in the same way?
> 
> That looks promising. Maybe someone with the dev environment setup should try inserting the library at this location: https://github.com/ferdium/ferdium-app/blob/develop/src/index.ts#LL77C5-L77C7
> 
> The code would be this one: https://github.com/ramboxapp/electron-chrome-extensions/blob/master/README.md#advanced
> 
> Hopefully it's that simple. But I doubt it because ferdium has many custom behaviors.

i have try 3 week agao, seems not work. But i can re-check when i have a little bit more time.

**Comment by RJVB** [comment]
FWIW, I just received a notification that "Stack" now supports all Chrome extensions (as a beta feature).

I won't be using it though; the interface is so gaudy that it bogs down my machine and already managed to crash the i915 driver (embedded, lower-end Intel GPU).

**Comment by R2Pitou** [comment]
Adding myself to the cacophony. Being able to use Chrome Extensions like 1password, AI ghostwriters etc. would bring a massive productivity boost to ferdium!

**Comment by fenglengshun** [comment]
There's already so many people asking for this, but how _would_ this be implemented? Isn't Ferdium using electron? Not sure if that supports extensions, beyond some .js ones (though, having a better UX for that would still help). Wavebox, which I'm currently use it. implement it by just being an actual browser - it's actual real Chromium browser with a different front-end a la Vivaldi. That would require a major rewrite of the project, though, in which case I get why there's no answer/plan here yet from the devs.

**Comment by nickchomey** [comment]
Agreed - this would be fantastic

**Comment by germain-italic** [comment]
+1 for Chrome extensions (not necessarily adblockers, I have a bunch of Gmail exts that enhance my workflow)

**Comment by D3SOX** [comment]
Please stop the +1 comments and just use the upvote function. I've subscribed to this issue for actual updates...

**Comment by Qu1ckJ** [comment]
+1 for Chrome extensions

**Comment by o0-o** [comment]
Are Chrome extensions at all feasible for a platform based on Electron? Looking at Rambox, it looks like they transitioned their whole platform from Electron to pure Chromium at which point they were able to support Chrome extensions. Of course, that was also the point at which they transitioned away from open source, so I suppose we can't know exactly what happened with the code.

Anyway, I just wanted to point out that implementing general support for Chrome extensions would likely involve a fundamental restructure of either Ferdium or Electron. Correct me if I'm wrong.

**Comment by vraravam** [comment]
> Anyway, I just wanted to point out that implementing general support for Chrome extensions would likely involve a fundamental restructure of either Ferdium or Electron

Yes, you are absolutely correct. Electron (as a framework) needs to provide support for chrome extensions, after which, based on volunteers in Ferdium, we can try to incorporate/enable the same. Without either of them, its pointless to keep adding +1 messages to this ticket.
For those who want to watch/subscribe to this thread, please click on the "Subscribe" button on the right, rather than adding a comment. It only creates spam for those of us watching, without any positive impact on the outcomes. TIA

**Comment by Julian88Tex** [comment]
Thanks for the extra context. RE electron supporting chrome extensions, is there perhaps an electron issue we can subscribe to with this request?

RE +1s, I suspect many people don't know or forget about this github feature. Seems like a reasonable Github feature request is for it to prompt and subscribe or emoji response before posting.

**Comment by o0-o** [comment]
> RE electron supporting chrome extensions, is there perhaps an electron issue we can subscribe to with this request?

If that issue exists, linking it here and freezing additional comments (assuming that is possible?) would make sense, imho.

**Comment by Julian88Tex** [comment]
It appears Electrons documentation explicitly states that supporting extensions is a "non-goal": https://electronjs.org/docs/latest/api/extensions

In that case it seems like this should be closed as a won't do with some reasoning provided or it should be linked to another issue which discusses the possibility of rewriting Ferdium as a non-Electron app with pros and cons.

**Comment by martinszy** [comment]
What about the option we were talking about last year with the rambox code? 

here: https://github.com/ferdium/ferdium-app/issues/135#issuecomment-1566886105

**Comment by o0-o** [comment]
> What about the option we were talking about last year with the rambox code?
> 
> here: https://github.com/ferdium/ferdium-app/issues/135#issuecomment-1566886105

For anyone new to this issue, please hold any "+1" or other non-substantive posts until Ferdium devs can respond to @martinszy

**Comment by vraravam** [comment]
> here: [#135 (comment)](https://github.com/ferdium/ferdium-app/issues/135#issuecomment-1566886105)

This is non-trivial (and requires an almost full rewrite of Ferdium), and unless we have new volunteers who are willing to take this up, it won't be implemented. (I have greatly minimized my contribution to Ferdium, so I might not have the latest thinking from the rest of the core contributors - so please take that into account.)

**Comment by MurzNN** [comment]
As I see here
https://github.com/ferdium/ferdium-app/blob/55a5d3843f099d331866f818f5cc2b1b9b3b1ae1/src/index.ts#L546-L547
We're already creating a new Browser Windows for each service, so seems we can pass it to the extension from the example https://github.com/ramboxapp/electron-chrome-extensions/blob/master/README.md#advanced:
```
// Adds the active tab of the browser
  extensions.addTab(browserWindow.webContents, browserWindow)
```
Or this will not work and we shold rework the whole approach?

**Comment by vraravam** [comment]
where are extensions managed from? What format will that be "understood" in? do we want universally enabled or per-service-enabled? what about performance hits due to extensions?

**Comment by MurzNN** [comment]
> where are extensions managed from? What format will that be "understood" in?

As I see in this example https://github.com/samuelmaddock/electron-browser-shell - seems we should manually "pack" them: https://github.com/samuelmaddock/electron-browser-shell/tree/master/extensions
```
Place unpacked extensions (not .crx archives) here to have them automatically loaded by the browser.
```
After this, I guess they should automatically appear in the url bar at the right.

> do we want universally enabled or per-service-enabled? 

I think global enabling would be more convenient, especially for password managers like Bitwarden and ad blockers.

> what about performance hits due to extensions?

I think like in the usual Chrome instance, but I think everyone thirsty for this feature will be ready to get some performance degradation.

P.S. I understand that this is maybe not as trivial as I describe, but just want to give it a try, maybe the integration will be really not as hard as it looks. Especially if we already have a working example here https://github.com/samuelmaddock/electron-browser-shell

I'm not an Electron developer, so can't give any deep technical help from my side, but really wish to get extensions in the Ferdium, so ready to test and give ideas.

**Comment by RJVB** [comment]
How big of an issue is extension support in the Electron world? Supposing there's enough demand for it from dependent project devs who have the general know-how, wouldn't it be more efficient to fork Electron? If not only to streamline this `"Place unpacked extensions (not .crx archives) here to have them automatically loaded by the browser."` method?

**Comment by quentingosset** [comment]
> > where are extensions managed from? What format will that be "understood" in?
> 
> As I see in this example https://github.com/samuelmaddock/electron-browser-shell - seems we should manually "pack" them: https://github.com/samuelmaddock/electron-browser-shell/tree/master/extensions
> 
> ```
> Place unpacked extensions (not .crx archives) here to have them automatically loaded by the browser.
> ```
> 
> After this, I guess they should automatically appear in the url bar at the right.
> 
> > do we want universally enabled or per-service-enabled?
> 
> I think global enabling would be more convenient, especially for password managers like Bitwarden and ad blockers.
> 
> > what about performance hits due to extensions?
> 
> I think like in the usual Chrome instance, but I think everyone thirsty for this feature will be ready to get some performance degradation.
> 
> P.S. I understand that this is maybe not as trivial as I describe, but just want to give it a try, maybe the integration will be really not as hard as it looks. Especially if we already have a working example here https://github.com/samuelmaddock/electron-browser-shell
> 
> I'm not an Electron developer, so can't give any deep technical help from my side, but really wish to get extensions in the Ferdium, so ready to test and give ideas.

Hey @MurzNN do you have find something to make extension work ?
it's seems electron can use chrome extension. If you see the dependencies used on this product : https://codecanyon.net/item/web3-browser-ifox/46329836 you can see "electron-chrome-extensions": "^3.10.1". So i think we have a proof thats it possible. now, we need just learn how to use it and make it work :D

**Comment by kteppris** [comment]
This feature would be really need. Has there been any progress? :)

**Comment by vraravam** [comment]
waiting for new contributors to help with the backlog @kteppris . till then, no progress can be made

**Comment by RJVB** [comment]
Not that it makes a lot of difference with the backlog, but part of the reason behind the request for extension support is to be able to run an adblocker.

I can't recall if it's been suggested already, but adblocking could also be built in. Quite a few applications provide implementations of that which are compatible with standard AdBlock filter lists (I was just reminded it exists, or existed, in KMail, for instance).

I would be really surprised if no module exists for NodeJS and/or Electron that implements such functionality (and that would thus probably be a lot easier to integrate).

**Comment by fenglengshun** [comment]
Adblock and a GUI userscript manager would be enough in most cases IMO. I'd love to have more, but at that point it'd be a browser, and I do actually have actual browsers for that.

**Comment by RJVB** [comment]
On Sunday March 24 2024 23:20:05 Bayazid B. Halim wrote:
>a GUI userscript manager

Something like TamperMonkey?

That'd be great, but I have the impression that would take us more than halfway to full extension support. Many of the userscripts I are crippled if not downright broken if they don't have access to a settings store mechanism and a standardised way to put up a GUI.

But since we're talking userscripts ... anyone here have an idea how to get one of the existing video-autoplay inhibitor scripts to work? Most do NOT require settings access or a GUI, from what I can tell.

**Comment by MurzNN** [comment]
> Adblock and a GUI userscript manager would be enough in most cases IMO. I'd love to have more, but at that point it'd be a browser, and I do actually have actual browsers for that.

But we already have some kind of userscripts via editing the per-serivce `user.js` file from the service settings page. You can include there any hacks you want :)  But copy-pasting some kind of adblock there will be a pretty tricky task

**Comment by RJVB** [comment]
>But we already have some kind of userscripts via editing the per-serivce `user.js` file from the service settings page. 

Yes, but that only allows you to have a single script, so you would have to implement a multiplexer yourself to run multiple scripts. Plus the standard APIs browser extensions can use to put up a settings page, menus and store settings are missing.

**Comment by trymeouteh** [comment]
This would be an amzing feature to have

**Comment by odnar-dev** [comment]
i think this may be related : https://github.com/samuelmaddock/electron-browser-shell

**Comment by cobbinaloic** [comment]
Still hoping to see that feature in the future.

**Comment by pepper3k** [comment]
omg this sounds super complicated to implement, don't get your hopes up, we need more devs

**Comment by davidkennedydev** [comment]
**Maintainers**, do you intend to support and use a project like https://github.com/getstation/electron-chrome-extension to solve that issue?

**Comment by cobbinaloic** [comment]
I think that is an interesting idea. How do I support?

**Comment by TunaCuma** [comment]
Maybe some essential plugins can be made built-in like adblockers, vim shortcuts (like vimium) etc.

**Comment by Albirew** [comment]
actually, that's what rambox is doing in their pro edition: a few addons available like µblock origin, dark reader, bitwarden, etc.

**Comment by cobbinaloic** [comment]
I have tried Rambox in the past. I was not impressed.

**Comment by l4kr** [comment]
I like Rambox but I would pay like $5, not $210.
Either way I don't understand why Ferdium doesn't have this feature, it's just a Chromium browser?

**Comment by katastrophe92** [comment]
Does User Scripts work?

**Comment by quentingosset** [comment]
Honnestly, i use wavebox now. it's not free but all chrome extension is available.

**Comment by RJVB** [comment]
>Honnestly, i use wavebox now. it's not free but all chrome extension is available.

But to what extent is this still different from just running the services in a browser tab?

**Comment by jurgenhaas** [comment]
> Honnestly, i use wavebox now. it's not free but all chrome extension is available.

>> But to what extent is this still different from just running the services in a browser tab?

Perfect question. And that's why I banned all similar applications from my tool stack after I had moved maybe 5-6 times in a decade. In the end, it always turned into some disappointment. Now, I'm using Vivaldi as the replacement. It's fast, lightweight, and well maintained. My regular browser for browsing the web remains Firefox. That combination serves me well for the last 12 months.

**Comment by RJVB** [comment]
On Friday August 16 2024 07:08:55 Jürgen Haas wrote:
>Now, I'm using Vivaldi as the replacement. It's fast, lightweight, and well maintained. 

I think every ChromeClone claims at least the 1st two aspects, but I wasn't impressed by Vivaldi in either department when I tried it a few years back. Maybe if you actually exploit all its customisation features that you can make something that's not an eyesore but rather lean-and-mean. But I'd be surprised if you didn't have to redo that all over gain regularly after some more-or-less-major update.

In terms of "lean and mean" Chromium there's Flashpeak Slimjet, but I'm not certain how well maintained it is.

**Comment by edermachado** [comment]
I would like to have Languagetool.org integrated into Ferdium or at least as a plug-in option.
I use chat and email apps a lot, and languagetool.org helps me a lot in writing better texts.
The Chrome plug-in works very well, as does the desktop app, which integrates with several other apps, except Ferdium.

**Comment by RJVB** [comment]
On Monday April 21 2025 13:14:02 Éder Magalhães Machado wrote:

>I use chat and email apps a lot, and languagetool.org helps me a lot in writing better texts.

Interesting, but what a pity they didn't throw in an integrated translator!

**Comment by taro-ball** [comment]
Some kind of adblock is absolutely essential, until then I'll just keep using my opera sidebar for messengers. Edge's sidebar is also good, but fuck Microsoft.

**Comment by SHU-red** [comment]
> Maybe some essential plugins can be made built-in like adblockers, vim shortcuts (like vimium) etc.

Using `tridactyl` would be very nice

**Comment by ikkqqstudio** [comment]
3 years and Still no Chrome Extension? what they are doing?

**Comment by an0key** [comment]
> 3 years and Still no Chrome Extension? what they are doing?

You should probably ask for a refund... oh wait...

They accept PRs I believe.

**Comment by m-shazly** [comment]
we want to to add ad block extensions people

**Comment by proboner** [comment]
I was thinking ad blocking would be a fantastic addition, even if other extension support wasn't added. Ads are out of hand.

**Comment by ylluminate** [comment]
Anyone with macOS, just take the plunge and buy Little Snitch. Been using it for a very long time and it's fantastic - and allows you to block ads. I run Ferdium through a `FerdiumLauncher` with:
```
CHROMIUM_FLAGS=(
    "--disable-features=EnableQuic"
    "--disable-features=EncryptedClientHello"
)
```
So that I can control things a bit better in these regards.

**Comment by RJVB** [comment]
On Friday December 26 2025 18:47:06 ylluminate wrote:
>Anyone with macOS, just take the plunge and buy Little Snitch.

You can probably do much the same for free with privoxy, if you're up to setting it up as an https proxy.

**Comment by ylluminate** [comment]
True @RJVB - or probably many other tools like Pi-hole, etc. for essentially any topology. I suppose I was thinking "minimum viable product" optionality here wrt simplicity and a tool I already have in place and use regularly both on and off my controlled network... but I suppose privoxy could be installable on a localhost as well readily, just perhaps a bit more complexity in setup. Interesting though for sure! Thanks for noting it.

**Comment by RJVB** [comment]
On Saturday December 27 2025 12:57:58 ylluminate wrote:
> I suppose privoxy could be installable on a localhost as well readily

Prixovy is supposed to run as a proxy on the local host and by default will serve only that host.

It can do more than just blocking things too; I used to use it to change certain colours for instance.

**Comment by JackMBurch** [comment]
Yeah, for all intents and purposes ferdium is unusable without this feature... We are in 2026 where password managers are a must. If you can't autofill passwords for when you're frequently logged out from your services, it makes the app unusable. Love what you guys are doing, but this is a must.

I'd be interested in contributing to this if you guys don't have the bandwidth, but would need some guidance on where to start if anyone has any pointers.

**Comment by MurzNN** [comment]
@JackMBurch glad that you are ready to try! Probably using this package https://www.npmjs.com/package/electron-chrome-extensions it may work, could you try it?

**Comment by pepper3k** [comment]
@JackMBurch You have my sword ⚔️  Let me know what to do.

**Comment by Habermann76** [comment]
languagetool and a password manager  seems to be a very good idea
i would like tto use wildhero with ferdium to plant some trees while sending emails
but as password manager i maybe would like my google account instead of Lastpass
i once lost access to lastpass and had many following problems trying to remember where to change passwords and how
google password manager would be best or maybe a free pw manager that imports from chrome

**Comment by rulrok** [comment]
I like to keep chatgpt as a tab, but it gets extremelly sluggish when working with long prompts; there is a chrome extension I'd like to install but I cannot so I have to resort using on my regular browser; I like the possibility of installing chrome extension

**Comment by SCIFENEFICS** [comment]
Landed here because I also find it extremely frustrating having to open a browser, go to Bitwarden, find the password, and then come back just to log in again.

If Ferdium could have its own simple password manager, or even better, integrate with Bitwarden, that would be fantastic. I really only need to store 5–6 passwords, as I only use a handful of services in Ferdium for work.

Other than that, Ferdium is amazing for work!  Where i am constantly jumping between the same few services again and again.
Anyone figured out a good solution to this?

**Comment by ashwinvis** [comment]
My solution was to switch to regular PWA supported by Chromium, Brave etc. But I agree it would have nice to have extensions

**Comment by RJVB** [comment]
>ashwinvis left a comment (ferdium/ferdium-app#135)
>
>My solution was to switch to regular PWA supported by Chromium, Brave etc. But I agree it would have nice to have extensions 

From what I've seen, PWAs run in a browser session, typically your main/default session. That means you can't quit them without quitting your browsing session.

A better and IMHO easier emulation would be to set up a dedicated browser profile, and have your different services in tabs. You lose inter-service sandboxes (I don't think I really need those), the hibernation-with-periodic-wakeup feature and the possibility to "close" to the systray (which in my experience is a great way to reduce F's CPU load as much as possible without quitting it), and it would be too easy to close tabs accidentally.
